### PR TITLE
feat: added new component (small back button) as reusables.

### PR DESCRIPTION
**Pull request by dennisanthonyblabla**
Added new color set for the said componenet.

### CODE REVIEW

File: MacroChallengeTeam2/ViewController/ViewController.swift [hunk 1]
```diff
@@ -10,22 +10,31 @@ import SwiftUI
...
+        addActionToButton()
     }
     private func setupAutoLayout() {
-        profileUser.snp.makeConstraints { (make) in
+        smallBackButton.snp.makeConstraints { (make) in
             make.center.equalToSuperview()
         }
     }
+
+    func addActionToButton() {
```
Comment (m-salman-a) [inline MacroChallengeTeam2/ViewController/ViewController.swift]: ini in my opinion isi function ini langsung ditaro di viewdidload aja kalo cuma 1 line & dipake 1 kali, what do you think?

File: MacroChallengeTeam2/ViewController/ViewController.swift [hunk 2]
```diff
@@ -10,22 +10,31 @@ import SwiftUI
...
+        smallBackButton.addTarget(self, action: #selector(clickBack), for: .touchUpInside)
+    }
+    @objc func clickBack() {
```
Comment (m-salman-a) [inline MacroChallengeTeam2/ViewController/ViewController.swift:33]: kasih todo ketika ada behavior yang belom ada & bakal ada di kemudian hari
// TODO: add behavior when clicked button

File: MacroChallengeTeam2/View/SmallBackButtonView.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import UIKit
+import SnapKit
+
+final class SmallBackButtonView: UIButton {
+    enum Variants {
+        case variant1
+        case variant2
+        case variant3
+    }
+    var variant: Variants = .variant1 {
```
Comment (m-salman-a) [inline MacroChallengeTeam2/View/SmallBackButtonView.swift:20]: if i remember setelah } kayanya ada enter dulu sekali baru lanjut statement baru, coba tolong cek dari SwiftLint lagi

enum {
...
}

var variant: ...

File: MacroChallengeTeam2/View/SmallBackButtonView.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            case .variant3:
+                setImage(buttonImage?.withTintColor(.kobarBlueActive, renderingMode: .alwaysOriginal), for: .normal)
+                self.frontBG.backgroundColor = .white
+                self.backBG.backgroundColor = .white
+                self.backBG.layer.opacity = 0.7
+                self.alpha = 0.7
+            }
+        }
+    }
+    private lazy var frontBG: UIView = {
```
Comment (m-salman-a) [inline MacroChallengeTeam2/View/SmallBackButtonView.swift:48]: idem

**Review by m-salman-a — APPROVED** [review]
Approve with minor style changes, we can start a discussion if you disagree with any of the points

File: MacroChallengeTeam2/ViewController/ViewController.swift (hunk 1 above)
Comment (dennisanthonyblabla) [reply MacroChallengeTeam2/ViewController/ViewController.swift]: oke

**Pull request merged**